dotnet/src/AutoGen/OpenAI/Extension/MessageExtension.cs

```diff
@@ -162,4 +163,190 @@ public static ChatRequestFunctionMessage ToChatRequestFunctionMessage(this Messa
 
         return functionMessage;
     }
+
+    public static IEnumerable<ChatRequestMessage> ToOpenAIChatRequestMessage(this IAgent agent, IMessage message)
```

Comment:
Maybe we need to break it into small pieces

Reply:
Code-wise we can break it up into a chain of message handlers. Though functionality-wise we don't need to do it now. Unless you have some other considerations?

Reply:
This might be a better implementation but yes, functionality-wise no need to change that code. Other than we can probably make it internal (no need to expose this as a public method)
https://github.com/microsoft/autogen/pull/1676/files#diff-99bde4c9640d0c773927d1462185e5bca0dae555ad22df353645469b755c921aR164